### PR TITLE
Swap out KeyTransform for CaseTransform

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
#### Purpose

To allow key transform to be swapped out with native extensions for performance boosts. See: https://github.com/NullVoxPopuli/case_transform-rust-extensions (Up to 28 times faster)

#### Changes

- Delete KeyTransform
- Use CaseTransform Gem

#### Caveats

None

#### Related GitHub issues

#1834?

#### Additional helpful information

Benchmarks (Ruby only):

This PR:
```
camel 8637.508094079541/ips; 189 objects
camel_lower 8407.325157949488/ips; 189 objects
dash 6564.254917142684/ips; 189 objects
unaltered 7704142.353092194/ips; 1 objects
underscore 6792.223670331557/ips; 189 objects
```

Master:
```
camel 644.5149854118894/ips; 3653 objects
camel_lower 729.6670949346017/ips; 2622 objects
dash 3320.993361425401/ips; 770 objects
unaltered 7361102.918352639/ips; 1 objects
underscore 6463.657823115512/ips; 313 objects
```

The performance boost here is due to CaseTransform utilizing a cache.

### CODE REVIEW

File: active_model_serializers.gemspec [hunk 1]
```diff
@@ -43,6 +43,7 @@ Gem::Specification.new do |spec|
   # 'thread_safe'
 
   spec.add_runtime_dependency 'jsonapi', '0.1.1.beta2'
+  spec.add_runtime_dependency 'case_transform', '>= 0.2'
```
Comment (bf4) [inline active_model_serializers.gemspec]: I'd like to see this moved to the rails-api organization first.  Thoughts?

Comment (NullVoxPopuli) [inline active_model_serializers.gemspec]: I'm cool with that. 
should we email github to just assign the existing repo to rails-api?

Comment (NullVoxPopuli) [inline active_model_serializers.gemspec]: done

**Comment by NullVoxPopuli** [comment]
rebased

**Pull request merged**